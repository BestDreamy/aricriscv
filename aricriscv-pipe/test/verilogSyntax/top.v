`timescale 1ns / 1ps
`include "./low.v"

module top;

    reg clk;
    reg[3:0] out1 = 1;
    wire[3:0] out2;
    always @(posedge clk) begin
        out1 <= 0;
    end

    low low_ins (
        .clk(clk),
        .out1(out1),
        .out2(out2)
    );

    initial begin
        clk = 0;
        $dumpfile("./wave.vcd");
        $dumpvars;
        #100 $finish;
    end

    always #3 clk = ~clk;

endmodule