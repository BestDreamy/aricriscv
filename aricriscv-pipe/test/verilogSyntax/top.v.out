#! /usr/local/iverilog/bin/vvp
:ivl_version "11.0 (devel)" "(s20150603-642-g3bdb50da)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 12;
:vpi_module "system";
:vpi_module "vhdl_sys";
:vpi_module "vhdl_textio";
:vpi_module "v2005_math";
:vpi_module "va_math";
S_00000000010c7210 .scope module, "top" "top" 2 4;
 .timescale -9 -12;
v00000000011159b0_0 .var "clk", 0 0;
v0000000001115a50_0 .var "out1", 3 0;
L_0000000001157c38 .functor BUFT 1, C4<0010>, C4<0>, C4<0>, C4<0>;
RS_0000000001117038 .resolv tri, v00000000010ca6b0_0, L_0000000001157c38;
v0000000001115af0_0 .net8 "out2", 3 0, RS_0000000001117038;  2 drivers
S_0000000001115820 .scope module, "low_ins" "low" 2 13, 3 1 0, S_00000000010c7210;
 .timescale -9 -12;
    .port_info 0 /INPUT 1 "clk";
    .port_info 1 /INPUT 4 "out1";
    .port_info 2 /OUTPUT 4 "out2";
v00000000010c73a0_0 .net "clk", 0 0, v00000000011159b0_0;  1 drivers
v00000000010ca610_0 .net "out1", 3 0, v0000000001115a50_0;  1 drivers
v00000000010ca6b0_0 .var "out2", 3 0;
E_00000000010cce70 .event posedge, v00000000010c73a0_0;
    .scope S_0000000001115820;
T_0 ;
    %wait E_00000000010cce70;
    %load/vec4 v00000000010ca610_0;
    %assign/vec4 v00000000010ca6b0_0, 0;
    %jmp T_0;
    .thread T_0;
    .scope S_00000000010c7210;
T_1 ;
    %pushi/vec4 1, 0, 4;
    %store/vec4 v0000000001115a50_0, 0, 4;
    %end;
    .thread T_1;
    .scope S_00000000010c7210;
T_2 ;
    %wait E_00000000010cce70;
    %pushi/vec4 0, 0, 4;
    %assign/vec4 v0000000001115a50_0, 0;
    %jmp T_2;
    .thread T_2;
    .scope S_00000000010c7210;
T_3 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v00000000011159b0_0, 0, 1;
    %vpi_call 2 21 "$dumpfile", "./wave.vcd" {0 0 0};
    %vpi_call 2 22 "$dumpvars" {0 0 0};
    %delay 100000, 0;
    %vpi_call 2 23 "$finish" {0 0 0};
    %end;
    .thread T_3;
    .scope S_00000000010c7210;
T_4 ;
    %delay 3000, 0;
    %load/vec4 v00000000011159b0_0;
    %inv;
    %store/vec4 v00000000011159b0_0, 0, 1;
    %jmp T_4;
    .thread T_4;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "top.v";
    "././low.v";
